### DISCUSSION TITLE
How to use HasRenderLoop

### DISCUSSION THREAD

**Discussion by charlenni** [mono/SkiaSharp#1658] [Q&A]
If I use a SKGLView in SkiaSharp.Forms, then I have to set `HasRenderLoop` to true. If I don't do this, I don't get an image, even if I call `InvalidateSurface()`. 

Ideally I would like to draw the canvas once and then wait until an event occurs, that needs a redrawn of the canvas. I thought, that this would be possible with `HasRenderLoop` set to false.

So any help or information about flag `HasRenderLoop`?

**Comment by mattleibow** [comment]
That sounds like a bug. What platform is this?

Typically, if you don't have a render loop, it should still render on size changes and when you manually call invalidate. 

What version is this and can you attach a repro? Not rendering at all sounds very strange.

**Reply by charlenni** [reply]
Thank you for looking into this.

The used version is 2.80.2

I only have the whole project. You could find it [here](https://github.com/charlenni/VTMap). The `HasRenderLoop` setting could be found at VTMapView.cs in project VTMap.View.Forms on line 28. Start it on UWP with `HasRenderLoop` set to true to see the normal behavior. Then change to false. Nothing is drawn than.
